api/types.go

```diff
@@ -310,6 +310,16 @@ type CreateRequest struct {
 	Quantization string `json:"quantization,omitempty"`
 }
 
+type TemplateRequest struct {
+	Model    string    `json:"model"`
+	Messages []Message `json:"messages"`
+	Tools    []Tool    `json:"tools"`
+}
+
+type TemplateResponse struct {
+	TemplatedPrompt string `json:"templated_prompt"`
```

Comment:
Is this a new API for us? Can we all chat about this field name?

Reply:
New API yes - name is not final -  open to suggestions!